src/WebJobs.Script.WebHost/Middleware/HostAvailabilityCheckMiddleware.cs

```diff
@@ -20,25 +20,29 @@ public HostAvailabilityCheckMiddleware(RequestDelegate next, ILoggerFactory logg
 
         public async Task Invoke(HttpContext httpContext, WebHostResolver resolver)
         {
-            using (Logger.VerifyingHostAvailabilityScope(_logger, httpContext.TraceIdentifier))
+            var manager = resolver.GetWebScriptHostManager();
```

Comment:
Wish we were doing this on top of `hosting`. All of these calls will need to change

Reply:
Will put this on hold until DI changes are in. This PR serves it's purpose of agreeing on the high level design. The mechanics of rebasing onto DI are secondary.